src/Components/WebAssembly/Build/src/Tasks/BlazorGetFileHash.cs

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Security.Cryptography;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.Build.Tasks
+{
+    /// <summary>
+    /// Computes the checksum for a single file.
+    /// </summary>
+    public sealed class BlazorGetFileHash : Task
+    {
+        internal const string _defaultFileHashAlgorithm = "SHA256";
+        internal const string _hashEncodingBase64 = "base64";
+
+        /// <summary>
+        /// The files to be hashed.
+        /// </summary>
+        [Required]
+        public ITaskItem[] Files { get; set; }
+
+        /// <summary>
+        /// The algorithm. Allowed values: SHA256, SHA384, SHA512. Default = SHA256.
+        /// </summary>
+        public string Algorithm { get; set; } = _defaultFileHashAlgorithm;
+
+        /// <summary>
+        /// The metadata name where the hash is stored in each item. Defaults to "FileHash".
+        /// </summary>
+        public string MetadataName { get; set; } = "FileHash";
+
+        /// <summary>
+        /// The encoding to use for generated hashs. Defaults to "hex". Allowed values = "base64".
+        /// </summary>
+        public string HashEncoding { get; set; } = _hashEncodingBase64;
+
+        /// <summary>
+        /// The hash of the file. This is only set if there was one item group passed in.
+        /// </summary>
+        [Output]
+        public string Hash { get; set; }
+
+        /// <summary>
+        /// The input files with additional metadata set to include the file hash.
+        /// </summary>
+        [Output]
+        public ITaskItem[] Items { get; set; }
+
+        public override bool Execute()
+        {
+            if (!SupportsAlgorithm(Algorithm))
+            {
+                Log.LogError("UnrecognizedHashAlgorithm", Algorithm);
+                return false;
+            }
+
+            if (Log.HasLoggedErrors)
+            {
+                return false;
+            }
+
+            System.Threading.Tasks.Parallel.ForEach(Files, file =>
+            {
+                var hash = ComputeHash(Algorithm, file.ItemSpec);
```

Comment:
Is removing the `FileExists` check from the original source intentional here?

Reply:
I guess I should revive it. MSBuild's implementation used internal APIs and I ended up removing those.

Reply:
Though remember that File.Exists is a race condition and it may be better to just catch and react to a FileNotFoundException from opening the stream.